src/core/shape_inference/include/ov_optional.hpp

```diff
@@ -4,7 +4,13 @@
 
 #pragma once
 
-#include <cstddef>
+
+#ifdef OPENVINO_CPP_17_VER
```

Comment:
This is not required as CPP17 is default standard

Reply:
Does it mean that we also want to remove the check in L16?

Reply:
As we moved to CPP17 as default standard, this file could be even removed, and use `std::optional` instead `ov::optional` or just leave alias.

Reply:
removed in https://github.com/openvinotoolkit/openvino/pull/28949

Reply:
I'll mark this conversation as resolved once that PR got merged.